src/libraries/Microsoft.Extensions.Configuration.Abstractions/ref/Microsoft.Extensions.Configuration.Abstractions.cs

```diff
@@ -26,9 +26,11 @@ public static partial class ConfigurationPath
         public static readonly string KeyDelimiter;
         public static string Combine(System.Collections.Generic.IEnumerable<string> pathSegments) { throw null; }
         public static string Combine(params string[] pathSegments) { throw null; }
+        public static string CombineWith(string separator, params string[] pathSegments) { throw null; }
```

Comment:
Why would you need to add new APIs here? I wasn't expecting new API added for this issue.

Reply:
The new method is because the `ConfigurationSection` can now have separators that are not `:`.  The original `Combine` method assumed that it would always be `:`, which is no longer the case.  This is no longer the case as users can specify a different separator, e.g. `` ` ``, so that they can have key names with colons

Reply:
Is there anything else needed for this PR? It fixes the issue reported and doesn't break any existing functionality. There's new API's, but they're needed if we want key separators to be configurable.

Reply:
Would you be open to summarizing the result of this PR into the issue page Something with a structure similar to the description in https://github.com/dotnet/runtime/issues/15725? (containing Summary, Rationale, Usage, and Proposed API). Then let's close this PR.

Reply:
@maryamariyan - sure, https://github.com/dotnet/runtime/issues/67616

Please let me know if it can be improved.